ci_scripts/on_failure.sh

```diff
@@ -13,6 +13,10 @@ else
   build_dir=$TRAVIS_BUILD_DIR/$BUILD_DIR
 fi
 
+echo "Uploading ASTC test file"
+curl -T $repo_root/tests/cts/clitests/output/extract/png_2d/output_ASTC_4x4_UNORM_BLOCK_2D.png temp.sh > tmp.txt
```

Comment:
For the long term we need something like the below tar to gather all the failing images and upload them. This means the failing images need to be easily identifiable by name or directory.

Reply:
This is just a quick and dirty just for this scenario. I think we could improve a CTS test runner to output the list of files that are mismatching. And based on that list they could be packed up.

The filepath of the output images matches their tests, so it is easy to find which test generated them.
For example all the `output/extract/png_2d/output_*.png` file come from the `tests/extract/png_2d.json` test.

Reply:
I like the idea of the runner outputting (writing to a file) the list of mismatched files. We need this and modified on_failure.{ps1,sh} soon. As CI builds take so long we can't afford to have to hack `on_failure` and re-run to capture mismatches. At least use of `--output-failure` removes the need to upload LastTest.log.

Please revert to the current `on_failure.sh` for now then I'll merge this. Plan on updating the CTS runner and scripts as part of the rebase.

Reply:
I agree. The workarounds / debugging hacks have been already reverted yesterday.